applications/test_applications.py

```diff
@@ -0,0 +1,95 @@
+# ===------ test_applications.py ------------------------ *- Python -* ----=== #
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# ===----------------------------------------------------------------------=== #
+
+import os
+import re
+import sys
+from pathlib import Path
+parent = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(parent)
+
+from test_utils import *
+
+def setup_test():
+
+    return True
+
+def migrate_test():
+    src = []
+    extra_args = []
+    in_root = os.path.join(os.getcwd(), test_config.current_test)
+    test_config.out_root = os.path.join(in_root, 'out_root')
+
+    if test_config.current_test != 'llama':
+        for dirpath, dirnames, filenames in os.walk(in_root):
+            for filename in [f for f in filenames if re.match('.*(cu|cpp|c)$', f)]:
+                src.append(os.path.abspath(os.path.join(dirpath, filename)))
+
+    if test_config.current_test == 'llama':
+        original_string = '/export/users/placeholder/project/llama.cpp'
+        new_string = in_root
+        db_dir = os.path.join(in_root, 'gpubuild')
+        with open(os.path.join(db_dir, 'compile_commands.json'), "r") as file:
+            file_contents = file.read()
+        file_contents = file_contents.replace(original_string, new_string)
+        db_dir = db_dir.replace('\\', '\\\\')
+        file_contents = re.sub(r'"directory": ".*?"', '"directory": "' + db_dir + '"', file_contents)
+        file_contents = file_contents.replace('\\', '\\\\')
+        with open(os.path.join(db_dir, 'compile_commands.json'), "w") as file:
+            file.write(file_contents)
+        src.append(' -p=' + os.path.join(in_root, 'gpubuild'))
+        src.append(' --enable-profiling ')
+        src.append(' --use-experimental-features=free-function-queries,local-memory-kernel-scope-allocation ')
+
+    return do_migrate(src, in_root, test_config.out_root, extra_args)
+
+def build_test():
+    if (os.path.exists(test_config.current_test)):
+        os.chdir(test_config.current_test)
+    srcs = []
+    cmp_opts = []
+    link_opts = []
+    objects = []
+
+    llama_files = ['common.cpp', 'sampling.cpp.dp.cpp', 'console.cpp', 'grammar-parser.cpp', 'train.cpp.dp.cpp',
```

Comment:
How about use the --gen-build-script to generate the Makefile.dpct to build the migrated code?

Reply:
The --gen-build-script is not work well for llama.cpp, use --migrate-cmake-script is in progress.